index.d.ts

```diff
@@ -1,5 +1,9 @@
 declare module 'react-string-replace' {
-  function reactStringReplace(text: React.ReactNode, regex: RegExp, cb: (match: string, index: number) => string | JSX.Element): JSX.Element;
+  function reactStringReplace(
+    text?: string | React.ReactNodeArray, 
+    regex?: string | RegExp, 
+    cb?: (match: string, index: number, offset: number) => React.ReactNode
```

Comment:
The callback should return any `ReactNode`, which can also be an array or just a simple string.
Objects are not supported, since you could not use them as an React Element.

Reply:
Sounds reasonable to me. If you're accidentally returning an object you'd want to know